### DISCUSSION TITLE
XInput or Input Samples for XBox Controller

### DISCUSSION THREAD

**Discussion by 0w8States** [dotnet/Silk.NET#930] [General]
Hi!

Are there any example projects available for using Silk.Net.Input or Silk.Net.XInput for retrieving XBox controller inputs?

**Comment by roeyskoe** [comment]
Currently no, but listening for controller input is similar to keyboard or a mouse.
See: https://github.com/dotnet/Silk.NET/blob/main/examples/CSharp/OpenGL%20Tutorials/Tutorial%202.2%20-%20Camera/Program.cs#L108
Just get a controller from `input.Gamepads[index]`.

Contributions are welcome!
